annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1291,238 @@ def test_get_page_number_combinations(
+        users_default_load,
+        users_overall_load,
+    )
+    assert output == expected_output
+    for user, expected_share_load in zip(users, expected_share_loads):
+        assert user["share_load"] == expected_share_load
+
+
+@pytest.mark.parametrize(
+    ("files", "user_pages", "expected_output"),
+    (
+        (
+            [{"pages_number": 1, "file_id": 0}],
+            1,
+            [{"pages_number": 1, "file_id": 0}],
+        ),
+        (
+            [
+                {"pages_number": 10, "file_id": 0},
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+            90,
+            [
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+        ),
+    ),
+)
+def test_find_equal_files(
+    files: List[Dict[str, int]],
+    user_pages: int,
+    expected_output: List[Dict[str, int]],
+):
+    assert find_equal_files(files, user_pages) == expected_output
+
+
+@pytest.mark.parametrize(
+    ("files", "user_pages", "SPLIT_MULTIPAGE_DOC_setup", "expected_output"),
+    (
+        (
+            [{"pages_number": 10, "file_id": 0}],
+            20,
+            "",
+            ([{"pages_number": 10, "file_id": 0}], 0),
+        ),
+        (
+            [
+                {"pages_number": 10, "file_id": 0},
+                {"pages_number": 30, "file_id": 1},
+                {"pages_number": 0, "file_id": 2},
+            ],
+            20,
+            "a",
+            ([{"pages_number": 10, "file_id": 0}], 0),
+        ),
+    ),
+)
+def test_find_small_files(
+    files: List[Dict[str, int]],
+    user_pages: int,
+    SPLIT_MULTIPAGE_DOC_setup: str,
+    expected_output,
+):
+    def find_files_for_task_side_effect(files, pages_for_task):
+        lst = []
+        for file in files:
+            if file["pages_number"] and file["pages_number"] in pages_for_task:
+                lst.append(file)
+        return lst
+
+    with patch(
+        "annotation.distribution.main.SPLIT_MULTIPAGE_DOC",
+        SPLIT_MULTIPAGE_DOC_setup,
+    ), patch(
+        "annotation.distribution.main.find_files_for_task",
+        side_effect=find_files_for_task_side_effect,
+    ):
+        output = find_small_files(files, user_pages)
+        assert expected_output == output
+
+
+def test_prepare_response():
+    tasks = [{"user_id": 5}]
+    expected_output = tasks
+    mock_arg = Mock()
+
+    with patch("annotation.distribution.main.distribute", return_value=tasks):
+        assert (
+            prepare_response(
+                mock_arg,
+                mock_arg,
+                mock_arg,
+                mock_arg,
+                mock_arg,
+                mock_arg,
+                mock_arg,
+                mock_arg,
+            )
+            == expected_output
+        )
```

Comment:
what do we test here? seems like mock testing

Reply:
The function prepare_response only does:
return [ task for task in distribute(...)]
I didnt really understand why we have that function, since it returns the same list like distribute(...)
I only made this test to cover the function, but it really doesnt test any functionality